evalml/automl/automl_search.py

```diff
@@ -841,6 +844,18 @@ def _is_imbalanced(X, y, problem_type):
                 )
         self.exclude_featurizers = exclude_featurizers or []
 
+        if exclude_model_families:
```

Comment:
On the same line of thought as the potential issues around `allowed_model_families`, I think we need some handling here for `allowed_component_graphs` - probably that they shouldn't be able to be set at the same time